sklearn/feature_extraction/text.py

```diff
@@ -1078,7 +1078,7 @@ def fit(self, X, y=None):
 
             # log+1 instead of log makes sure terms with zero idf don't get
             # suppressed entirely.
-            idf = np.log(float(n_samples) / df) + 1.0
+            idf = np.log(float(n_samples) / df).astype(X.dtype) + 1.0
```

Comment:
I don't like this line:

`np.log(n_samples / df, dtype=X.dtype) + 1` seems less clutter to me. I don't know if df can be a scalar which in this case you might want to import division from future in this case

Reply:
Also X can be an `np.int*` in which case you should fall back float64 for backward compatibility, I think.

Reply:
Yes exactly! using `np.log(n_samples / df, dtype=X.dtype)` will raise a TypeError in case the datatype is int*. Using `np.log(float(n_samples) / df).astype(X.dtype)` will automatically fallback to np.float64 in case the conversion isn't possible

Reply:
So should I leave it that way or use if else to check if it's int* and use the definition given @glemaitre